pipes/WDL/workflows/tasks/assembly.wdl

```diff
@@ -95,7 +96,8 @@ task scaffold {
   Int?    nucmer_min_cluster
   Int?    scaffold_min_pct_contig_aligned
 
-  String  sample_name = basename(contigs_fasta, ".fasta")
+  # do this in multiple steps in case the input doesn't actually have "assembly1-x" in the name
+  String  sample_name = basename(basename(basename(contigs_fasta, ".fasta"), ".assembly1-trinity"), ".assembly1-spades")
```

Comment:
Not really 'cleaner,' but [`sub()`](https://software.broadinstitute.org/wdl/documentation/spec#string-substring-string-string) can probably be used here with a regex. Something like:

`String  sample_name = sub(contigs_fasta, "(?:\.assembly1\-spades)?(?:\.assembly1\-trinity)?\.fasta$", "")`
(not sure if the escaping is right for the WDL regex engine)

Reply:
Ooh good to know. As you note it'd probably require a little experimenting to make sure we get the syntax right with the escaping & string substitution.